### DISCUSSION TITLE
Perdita storico Energia dopo ripristino

### DISCUSSION THREAD

**Discussion by sky7176** [indomus/forum#1531] [HUB: Home Assistant]
Buon giorno,
ho dovuto re installare HA dopo un fault del disco SSD.
Ho restorato l'ultimo backup, ma con mia sorpresa mi sono perso tutto lo storico dei consumi energetici.
Avete idea di come si possano recuperare quei dati, non dovrebbero essere nel db sqlite?

Giuseppe

**Comment by snakuzzo** [comment]
Sì ma se il db si è corrotto, al primo start sarà stato ricreato. Vedi che forse avrà tenuto da parte quello corrotto...controlla sotto la stessa directory

**Reply by sky7176** [reply]
Grazie,
ma se faccio un restore non dovrebbe ripristinare anche il db?

**Reply by snakuzzo** [reply]
in teoria sì...apri il file zip del backup e vedi se c'è...

**Reply by sky7176** [reply]
Buon gorno,
ho posto il quesito sul forum di Home Assistant, mi hanno risposto, che si, il db viene inserito nel backup, ma il più delle volte l'operazione danneggia il db stesso, ne consegue che in fase di restore non viene recuperato e ne viene creato uno nuovo.
Mi hanno anche detto che ci stanno lavorando.

Questo è da tenere in considerazione in caso di installazione ex-novo.

Giuseppe

**Reply by sky7176** [reply]
Alla fine ho deciso di usare MariaDB come db di home assistant, per i backup li faccio con uno script esterno e risolvo il problema dei backup corrotti.

**Reply by sky7176** [reply]
Purtroppo nel processo di restore dei backup c'è sicuramente un baco.
Ho provato a re-installare su una seconda RP4 il backup di questa notte, dopo il riavvio, il componente Mariadb non era presente.
L'ho re-installato, lui ha creato il db vuoto, a questo punto mi sono restorato il db, ma i totalizzatori dell'energia si sono azzerati, anche se nel db restorato vedo tutti i dati.

Quello del restore dei backup sembra essere il tallone di achille per HA, se gli storici dei sensori mi interessano poco, gli storici dell'energia mi servono.

**Comment by calz64** [comment]
Ho fatto restore anch'io e i dati energia sono andati persi...., problema irrisolto, peccato.

**Comment by faustozago** [comment]
Ciao a tutti,
mi intrometto in questa discussione perche' a breve dovro' passare da raspberry a minipc, quindi fare il restore dal backup su drive.
C'e' qualche modo per scaricare/copiare manualmente (su pc o drive)  il database dei consumi energetici , per poi reinserirlo sempre manualemnte nella nuova istanza HA ?

Io per le piccole modifiche uso FILE EDITOR , e ho visto che si possono caricare i file,  ma non ho la minima idea di come si faccia a scaricarli , e sopratutto dove trovare questo/I database.

Grazie

**Reply by calz64** [reply]
Quello che ho capito io, ma non prenderlo per esatto, è il file del db che si corrompe durante il salvataggio e quindi al ripristino lo ignora e ne avvia uno nuovo. Non sapresi per il salvataggio, credo che un db lo devi esportare-importare, forse si può fare da riga di comando ma non saprei come.

**Reply by sky7176** [reply]
Io alla fine sono passato a Mariadb, il backup lo faccio con uno script esterno lanciato da un PC che ho sempre acceso in casa.
